src/pyk/proof/reachability.py

```diff
@@ -723,18 +722,25 @@ def step_proof(self) -> Iterable[StepResult]:
                 _LOGGER.warning(f'Bounded node {self.proof.id}: {curr_node.id} at bmc depth {self.proof.bmc_depth}')
                 return [APRProofBoundedResult(curr_node.id)]
 
+        # Terminal checks for current node and target node
         is_terminal = self.kcfg_explore.kcfg_semantics.is_terminal(curr_node.cterm)
-
         target_is_terminal = self.proof.target in self.proof._terminal
-        check_subsume = (target_is_terminal and is_terminal) or (not target_is_terminal)
+
+        terminal_result = [APRProofTerminalResult(node_id=curr_node.id)] if is_terminal else []
+
+        # Subsumption should be checked if and only if the target node
+        # and the current node are either both terminal or both not terminal
+        check_subsume = is_terminal == target_is_terminal
 
         if check_subsume:
```

Comment:
```suggestion
        if is_terminal == target_is_terminal:
```

This intermediate value is only used once, and the expression is not large.